### PR TITLE
feat:[#123] Rethinking the Contributors List

### LINKED ISSUE #123: Rethinking the Contributors List
The Contributors List is a page that showcase all of the contributors across Vanilla OS, currently, the pages isn't updated every week like it used to, with that, I was thinking while bringing the weekly indexing back, to also refresh the page with the idea of a mini [team page](https://vanillaos.org/team). I made a quick concept of how will something like this should look, my idea was like I said earlier an mini team page, the same idea, but with usernames, names and profile picture. And that will look something like this: 

image

I also have an idea in mind with a GitHub link under the profile picture, kind of like in the Team page, and maybe other links like website or X (Twitter) and others, as they can be fetched along side the others using the GitHub API. Others ideas are welcome :)

### PR DESCRIPTION
**Pull request by GabsEdits**
After about 3 weeks of rethinking the Contributors here is the new version of it! 
You can read the whole documentation of how we got to the design in #123. Here is a quick overview of what is changed rewrite of the page: First of all we have small cards for everyone, that when click link to the contributor's GitHub Profile, with that the have: the name, username and profile picture, I've made it so you can have any name, as long as you want, just it has a limit so you the name doesn't leave the cards, it has an ellipsis that appears when the name is close to the margin of the card, when you hover the name it will smoothly gets small and shows the full name. For the back stuff: I've used ~~Axios~~ `fetch` to get all the data that appears when you open the page. Same idea like the first version of this page: to get all the contributors across the organization and remove the duplicates. 
With that I want to thank everyone that told they're opinion and help me make this a better page :)

## To Do
- [x] Move for fetching from the GitHub API every time a user loads the page to a script that fetches the API every week, see [GabsEdits/fetchContributors](https://github.com/GabsEdits/fetchContributors) and [this commit](https://github.com/Vanilla-OS/website/pull/125/commits/e9a3abcb63ac09c0cda49f4c3b270b39c6c19230) (something might change).
- [x] Updated Mobile Styles for better UX
- [x] Filter Bots
- [x] Align Contributors Alphabetically

## Preview
You can see the latest change to the pull request on [pr.gabs.eu.org/team/contributors](https://pr.gabs.eu.org/team/contributors) or you can have a quick look of the page in light theme:

[Contributors List in Light Theme](https://github.com/Vanilla-OS/website/assets/110247388/cdb0da21-5629-4273-ad48-8f2c307c6ad6)

---
Closes #123

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   "dependencies": {
     "@types/marked": "^5.0.1",
     "@vueuse/head": "^1.1.26",
+    "axios": "^1.5.0",
```
Comment (mirkobrombin) [inline package.json]: Is Axios mandatory or can it be replaced by a standard `fetch()` call?

File: public/assets/css/components/contributors.css [hunk 2]
```diff
@@ -0,0 +1,85 @@
+.contributors {
```
Comment (mirkobrombin) [inline public/assets/css/components/contributors.css:1]: This file does not looks compatible with our CSS convention. Please take a look at this comment https://github.com/Vanilla-OS/website/pull/116#discussion_r1290500609

File: src/views/get-involved/Contributors.vue [hunk 3]
```diff
@@ -1,414 +1,75 @@
...
+  id: number;
+  name: string;
+  login: string;
+  avatar_url: string;
+  profile_url: string;
+}
 
 export default defineComponent({
-    name: 'contributors',
+  name: 'Contributors',
```
Comment (mirkobrombin) [inline src/views/get-involved/Contributors.vue]: `name` must be the same as the path defined in the router

**Review by mirkobrombin — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (GabsEdits) [reply package.json]: Based on my past experience in fetching API Data it's way better to use Axios then the fetch, and I am not sure if just fetch can manage to do the job, I will look into it

Comment (GabsEdits) [reply package.json]: I've looked more into it, and yes, it's possible just using `fetch()`, but it somehow seems that the rate limit gets exceeded way faster, I will look more into it.

Comment (GabsEdits) [reply package.json]: > .I've looked more into it, and yes, it's possible just using fetch(), ~~but it somehow seems that the rate limit gets exceeded way faster~~, I will look more into it.

It seems that it works just like Axios, so no issues, should I remove Axios and push the simple `fetch()` code and remove Axios?

Comment (mirkobrombin) [reply package.json]: Yes

Comment (GabsEdits) [reply package.json]: > Yes

All good, I've pushed the code

File: public/assets/css/components/contributors.css (hunk 2 above)
Comment (GabsEdits) [reply public/assets/css/components/contributors.css:1]: Can you please review the file again? I've made some changes

**Review by mirkobrombin — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]

**Comment by GabsEdits** [comment]
Is there anything else that needs to be changed/added?

**Review by kbdharun — APPROVED** [review]

**Review by kbdharun — CHANGES_REQUESTED** [review]
**Update**: As mentioned on Discord, This PR isn't ready for merge yet. We might need to discuss an alternative instead of fetching from the GitHub API everytime a user loads the page.

Also, Names would probably need to be fixed for Mobile users as it doesn't seem to be consistent throughout the page.

**Review by kbdharun — APPROVED** [review]
LGTM

**Pull request merged**